File: 409-longest-palindrome/longest-palindrome.java
class Solution {
    public int longestPalindrome(String s) {
        int longest = 0;
        Set<Character> set = new HashSet<>();
        for(int i=0; i<s.length(); i++){
            char ch = s.charAt(i);
            if(set.contains(ch)){
                set.remove(ch);
                longest += 2;
            }
            else
                set.add(ch);
        }
        return set.isEmpty()? longest:longest+1;
    }
}

File: 409-longest-palindrome/longest-palindrome.py
class Solution:
    def longestPalindrome(self, s: str) -> int:
        longest = 0
        charset = set()
        for ch in s:
            if ch in charset:
                charset.remove(ch)
                longest += 2
            else:
                charset.add(ch)
        if(charset):
            longest += 1
        return longest